src/core/midi/MidiPort.cpp

```diff
@@ -151,18 +151,36 @@ void MidiPort::processInEvent( const MidiEvent& event, const MidiTime& time )
 
 void MidiPort::processOutEvent( const MidiEvent& event, const MidiTime& time )
 {
-	// mask event
-	if( isOutputEnabled() && realOutputChannel() == event.channel() )
+	// If we selected a channel from 1-16, we will only route the selected channel.
+	// But if we selected channel 0 ("--") we will route all channels out.
+	if( isOutputEnabled() )
 	{
-		MidiEvent outEvent = event;
-
-		if( fixedOutputVelocity() >= 0 && event.velocity() > 0 &&
-			( event.type() == MidiNoteOn || event.type() == MidiKeyPressure ) )
+		// Selected channel 0 ("--"). Route all channels
+		if( outputChannel() == 0 )
 		{
-			outEvent.setVelocity( fixedOutputVelocity() );
+			MidiEvent outEvent = event;
+
+			if( fixedOutputVelocity() >= 0 && event.velocity() > 0 &&
+				( event.type() == MidiNoteOn || event.type() == MidiKeyPressure ) )
+			{
+				outEvent.setVelocity( fixedOutputVelocity() );
+			}
+
+			m_midiClient->processOutEvent( outEvent, time, this );
 		}
+		// Selected channel between 1-16. Mask event.
+		else if( realOutputChannel() == event.channel() )
```

Comment:
I don't see a difference between this block and the `outputChannel() == 0` one.

Reply:
That's true, should have written a single conditional `if( outputChannel() == 0 || realOutputChannel() == event.channel() )` instead, my bad. I'll change that!